test/test_asymetric.py

```diff
@@ -151,6 +153,25 @@ def test_RSA_OAEP(self):
 
         self.assertEqual(text, plainText)
 
+    def test_RSA_OAEPwithAAD(self):
+        if self.SoftHSMversion < 2:
+            self.skipTest("RSA OAEP only supported by SoftHSM >= 2")
+
+        if self.manufacturer.startswith("SoftHSM"):
+            self.skipTest("SoftHSMv2 returns CKR_ARGUMENTS_BAD. 'AAD' not in expected format or unsupported.")
```

Comment:
What is "AAD"?

Reply:
AAD is "Additional authentication data" (pSourceData of CK_RSA_PKCS_OAEP_PARAMS struct). SoftHSMv2 does not like what I pass. Maybe it expects a certain format or does not support it. I have tested it with a hardware device. It passes.